src/command_modules/azure-cli-extension/azure/cli/command_modules/extension/custom.py

```diff
@@ -0,0 +1,126 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import os
+import tempfile
+import shutil
+import logging
+
+import requests
+from wheel.install import WHEEL_INFO_RE
+
+from six import StringIO
+from six.moves.urllib.parse import urlparse  # pylint: disable=import-error
+
+from azure.cli.core.util import CLIError
+from azure.cli.core.extension import (extension_exists, get_extension_path, get_extensions,
+                                      get_extension, ExtensionNotInstalledException)
+
+import azure.cli.core.azlogging as azlogging
+
+
+logger = azlogging.get_az_logger(__name__)
+
+OUT_KEY_NAME = 'name'
+OUT_KEY_VERSION = 'version'
+OUT_KEY_TYPE = 'extensionType'
+OUT_KEY_METADATA = 'metadata'
+
+
+def _run_pip(pip, pip_exec_args):
+    log_stream = StringIO()
+    log_handler = logging.StreamHandler(log_stream)
+    log_handler.setFormatter(logging.Formatter('%(name)s : %(message)s'))
+    pip.logger.handlers = []
+    pip.logger.addHandler(log_handler)
+    # Don't propagate to root logger as we catch the pip logs in our own log stream
+    pip.logger.propagate = False
+    logger.debug('Running pip: %s %s', pip, pip_exec_args)
+    status_code = pip.main(pip_exec_args)
+    log_output = log_stream.getvalue()
+    logger.debug(log_output)
+    log_stream.close()
+    if status_code > 0:
+        raise CLIError('An error occurred. Use --debug for more information.')
```

Comment:
I suggest just log the `status_code` and the error output, so i don't need to always run the command again when the error is obvious